src/Bedrock.Framework.Experimental/Protocols/Memcached/MemcachedMessageWriter.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Bedrock.Framework.Infrastructure;
+using Bedrock.Framework.Protocols;
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Bedrock.Framework.Experimental.Protocols.Memcached
+{
+    public class MemcachedMessageWriter : IMessageWriter<MemcachedRequest>
+    {
+        public void WriteMessage(MemcachedRequest message, IBufferWriter<byte> output)
+        {           
+            Span<byte> headerSpan = (stackalloc byte[Constants.HeaderLength]).Slice(0, Constants.HeaderLength);
+            var extraLength = 0;
+            if (message.Opcode == Enums.Opcode.Set)
+                extraLength = 8;
+            var messageValue = 0;
+            if (message.Value != null)
+                messageValue = message.Value.Length;
+            var header = new MemcachedRequestHeader()
+            {                
+                KeyLength = (ushort)message.Key.Length,
+                Opaque = message.Opaque,
+                TotalBodyLength = (uint)(extraLength + message.Key.Length + messageValue),
+                ExtraLength = (byte)extraLength   
```

Comment:
You could init `extraLength` as `byte` then this cast is not necessary. Furthermore it's clear that `extraLength` won't overflow when the correct type is used.